include/common/oclengine.hpp

```diff
@@ -144,10 +144,10 @@ class OCLDeviceContext {
 
     OCLDeviceCall Reserve(OCLAPI call) { return OCLDeviceCall(mutex, calls[call]); }
 
-    std::vector<cl::Event> ResetWaitEvents()
+    std::vector<cl::Event>& ResetWaitEvents()
     {
-        std::vector<cl::Event> waitVec = wait_events;
-        wait_events.clear();
+        std::vector<cl::Event>& waitVec = wait_events;
```

Comment:
Unnecessary temp.

Reply:
How so? This is the entire point, in my understanding. The pass by reference might or might not be necessary, but the bug was that `cl::Event` objects have to be treated as unique, we have to avoid calling their destructors until we're 100% done with them, and we have to call their destructors exactly once.

Reply:
So we have to get the class attribute clear, with an empty list, and also pass back the original list.